### PR TITLE
Login page edition2

### PR DESCRIPTION
**Pull request by sepehr9877**
#79  i edit this functionality by adding HttpSession to Controller and add user info as an attribute inside it

### CODE REVIEW

**Review by SoroorAghamiri — DISMISSED** [review]
Let's move forward with http session. Apparently cache config was causing errors, and this one seems simpler and easier to use for this project.

**Review by SoroorAghamiri — DISMISSED** [review]

File: src/main/java/ca/concordia/soen6011/controller/AssignInterviewController.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
+package ca.concordia.soen6011.controller;
```
Comment (MajorTreble) [inline src/main/java/ca/concordia/soen6011/controller/AssignInterviewController.java:1]: Don't need so many controllers. You could have one controller for Employer which matches "/employer/". This makes it easier to find related queries.

File: src/main/java/ca/concordia/soen6011/controller/AssignInterviewController.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+@Controller
+@RequestMapping("/employer/")
+public class AssignInterviewController {
+
+	@Autowired
+	private CandidateRepository candidateRepository;
+	@GetMapping("browsecandidate")
+	private String browse_candidate(Model model) {
+		List<Candidate> candidates=this.candidateRepository.findAll();
```
Comment (MajorTreble) [inline src/main/java/ca/concordia/soen6011/controller/AssignInterviewController.java:22]: This is just a list of everyone. There should be search parameters like candidates who have applied or candidates with a specific degree.

File: src/main/java/ca/concordia/soen6011/controller/LoginController.java [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+ * otherwise, they are asked to re-enter their information.
+ * @author Soroor Seyed Aghamiri
+ */
+@Controller
+public class LoginController {
+    @Autowired
+    private UserRepository userRepository;
+
+    @Autowired
+    private ProfileController profileController;
```
Comment (MajorTreble) [inline src/main/java/ca/concordia/soen6011/controller/LoginController.java:28]: You don't need this.

File: src/main/java/ca/concordia/soen6011/controller/LoginController.java [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+    @PostMapping("/submit")
+    public ModelAndView login(@RequestParam("username") String userName,
+                              @RequestParam("password") String pass, Model model,HttpSession session) {
+        ModelAndView modelAndView = new ModelAndView();
+        User result = userRepository.findByUsernameAndPassword(userName, pass);
+
+        if (result != null) {
+        	//set username is our session
+        	session.setAttribute("username",result.getUsername());
+            String userType = userRepository.findUser_Type(userName);
```
Comment (MajorTreble) [inline src/main/java/ca/concordia/soen6011/controller/LoginController.java:54]: Not used.

File: src/main/java/ca/concordia/soen6011/enums/InterviewType.java [hunk 5]
```diff
@@ -0,0 +1,8 @@
+package ca.concordia.soen6011.enums;
+
+public enum InterviewType {
```
Comment (MajorTreble) [inline src/main/java/ca/concordia/soen6011/enums/InterviewType.java:3]: This should go where it is used.

File: src/main/java/ca/concordia/soen6011/repository/UserRepository.java [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+
+import ca.concordia.soen6011.model.User;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.query.Param;
+
+public interface UserRepository extends JpaRepository<User,Integer> {
+    public User findByUsernameAndPassword(String username, String password);
+
+    @Query(value="SELECT user_type FROM User WHERE username = :userName",nativeQuery = true)
```
Comment (MajorTreble) [inline src/main/java/ca/concordia/soen6011/repository/UserRepository.java:11]: Don't need this

**Review by MajorTreble — DISMISSED** [review]

**Review by MajorTreble — APPROVED** [review]

**Comment by MajorTreble** [comment]
Do testing or documentation

File: src/main/java/ca/concordia/soen6011/model/Interview.java [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+@Entity
+public class Interview {
+	@Id
+	@GeneratedValue(strategy = GenerationType.IDENTITY)
+	private int id;
+	
+	private LocalDate date;
```
[the remark this answers is not in the document]
Comment (sepehr9877) [reply src/main/java/ca/concordia/soen6011/model/Interview.java:19]: i put them into different column so when i am about to save time and date i can send them separately which is easier to show and save

File: src/main/java/ca/concordia/soen6011/controller/LoginController.java (hunk 4 above)
Comment (sepehr9877) [reply src/main/java/ca/concordia/soen6011/controller/LoginController.java:54]: we need it it is for login functionality to set username into session

Comment (MajorTreble) [reply src/main/java/ca/concordia/soen6011/controller/LoginController.java:54]: You don't need the usertype

Comment (sepehr9877) [reply src/main/java/ca/concordia/soen6011/controller/LoginController.java:54]: yes correct

File: src/main/java/ca/concordia/soen6011/controller/LoginController.java (hunk 3 above)
Comment (SoroorAghamiri) [reply src/main/java/ca/concordia/soen6011/controller/LoginController.java:28]: Yes I forgot to delete it! I will change it when I submit my test files.

File: src/main/java/ca/concordia/soen6011/controller/LoginController.java (hunk 4 above)
Comment (SoroorAghamiri) [reply src/main/java/ca/concordia/soen6011/controller/LoginController.java:54]: I'll remove that too.

**Review by SoroorAghamiri — APPROVED** [review]
I deleted the unused lines and updated the documentation.

**Review by sepehr9877 — COMMENTED** [review]
ok cool

**Review by SoroorAghamiri — DISMISSED** [review]

**Review by SoroorAghamiri — APPROVED** [review]

**Review by MajorTreble — DISMISSED** [review]
Big change

File: src/main/java/ca/concordia/soen6011/controller/AssignInterviewController.java (hunk 2 above)
Comment (sepehr9877) [reply src/main/java/ca/concordia/soen6011/controller/AssignInterviewController.java]: > This is just a list of everyone. There should be search parameters like candidates who have applied or candidates with a specific degree.

yes i know ,but i have to think about about to see what is the best for search parameter but for now employer has a list of all candidate and also list of candidate with application having interview or waiting to get an interview by employer . employer can assign interview to those candidates who have applied for application . this lists are separate into two different table

**Review by SoroorAghamiri — APPROVED** [review]

**Review by nioushamahjoub — APPROVED** [review]

**Review by nioushamahjoub — APPROVED** [review]

**Review by nioushamahjoub — APPROVED** [review]

**Review by nioushamahjoub — APPROVED** [review]

**Review by nioushamahjoub — APPROVED** [review]

**Review by nioushamahjoub — APPROVED** [review]

**Review by nioushamahjoub — APPROVED** [review]

**Review by nioushamahjoub — APPROVED** [review]

**Review by nioushamahjoub — APPROVED** [review]

**Review by nioushamahjoub — APPROVED** [review]

**Review by nioushamahjoub — APPROVED** [review]

**Review by nioushamahjoub — APPROVED** [review]

**Review by SoroorAghamiri — APPROVED** [review]

**Review by SoroorAghamiri — APPROVED** [review]

**Review by SoroorAghamiri — APPROVED** [review]

**Review by SoroorAghamiri — APPROVED** [review]

**Review by SoroorAghamiri — DISMISSED** [review]
All good!

**Review by MajorTreble — APPROVED** [review]

**Pull request closed without merging**